src/NServiceBus.Core/Unicast/BuilderExtensions.cs

```diff
@@ -1,21 +1,21 @@
 namespace NServiceBus.Unicast
 {
     using System;
-    using ObjectBuilder;
+    using Microsoft.Extensions.DependencyInjection;
 
     /// <summary>
-    /// Extension methods for <see cref="IBuilder" />.
+    /// Extension methods for <see cref="IServiceProvider" />.
     /// </summary>
     public static class BuilderExtensions
     {
         /// <summary>
         /// Applies the action on the instances of <typeparamref name="T" />.
         /// </summary>
-        public static void ForEach<T>(this IBuilder builder, Action<T> action)
+        public static void ForEach<T>(this IServiceProvider builder, Action<T> action)
```

Comment:
we should just obsolete this method. it doesn't seem to be used anymore but it's public. I can't see why it would be NServiceBus' responsibility to bring in such an extension method?
(can be done is a separate PR though)

Reply:
Added to the to-do list.